internal/pkg/cli/pipeline_init.go

```diff
@@ -8,93 +8,244 @@ import (
-	// 		return err
-	// 	}
-	// }
-	// if opts.GitHubAccessToken == "" {
-	// 	if err := opts.askGitHubAccessToken(); err != nil {
-	// 		return err
-	// 	}
+	if len(opts.Environments) == 0 {
+		if err := opts.selectEnvironments(true); err != nil {
+			return err
+		}
+	}
+
+	if opts.GitHubRepo == "" {
+		if err := opts.selectGitHubRepo(); err != nil {
+			return err
+		}
+	}
+
+	if opts.GitHubAccessToken == "" {
+		if err := opts.getGitHubAccessToken(); err != nil {
+			return err
+		}
+	}
+
+	// if err := opts.askEnableCD(); err != nil {
+	// 	return err
 	// }
 
-	// if len(opts.Environments) == 0 {
-	// 	if err := opts.askEnvironments(); err != nil {
-	// 		return err
-	// 	}
+	// TODO ask this after pipeline.yml is written
+	// if err := opts.askDeploy(); err != nil {
+	// 	return err
 	// }
+
 	return nil
 }
 
 // Validate returns an error if the flag values passed by the user are invalid.
 func (opts *InitPipelineOpts) Validate() error {
 	// TODO
+	if viper.GetString(projectFlag) == "" {
+		return errNoProjectInWorkspace
+	}
 	return nil
 }
 
 // Execute writes the pipline manifest file.
 func (opts *InitPipelineOpts) Execute() error {
-	// TODO
+	opts.manifestPath = "pipeline.yml" // TODO: placeholder
+
+	log.Infoln()
+	log.Successf("Wrote the pipeline for %s app at '%s'\n", color.HighlightUserInput(opts.GitHubRepo), color.HighlightResource(opts.manifestPath))
+	log.Infoln("Your pipeline manifest contains configurations for your CodePipeline resources, such as your pipeline stages and build steps.")
+	log.Infoln()
 	return nil
 }
 
-func (opts *InitPipelineOpts) askAppName() error {
-	// projectName := viper.GetString(projectFlag)
-	// apps, err := opts.appStore.ListApplications(projectName)
-	// if err != nil {
-	// 	return err
-	// }
+func (opts *InitPipelineOpts) selectEnvironments(addMore bool) error {
+	if addMore == false {
+		return nil
+	}
 
-	// selections := make([]string, len(apps))
-	// for _, app := range apps {
-	// 	selections = append(selections, app.Name)
-	// }
+	addEnv, err := opts.prompt.Confirm(
+		pipelineAddEnvPrompt,
+		"Adds an environment that corresponds to a deployment stage in your pipeline. Environments are added sequentially.",
+	)
 
-	// sel, err := opts.prompt.SelectOne(
-	// 	fmt.Sprintf("Which application would you like to create a pipeline for?"),
-	// 	`The application to create a pipeline for.`,
-	// 	selections,
-	// )
-	// if err != nil {
-	// 	return fmt.Errorf("failed to select application: %w", err)
-	// }
+	if err != nil {
+		return fmt.Errorf("failed to confirm adding an environment: %w", err)
+	}
+
+	var selectMoreEnvs bool
+	if addEnv {
+		selectMore, err := opts.selectEnvironment()
+		if err != nil {
+			return err
+		}
+		selectMoreEnvs = selectMore
+	}
+
+	return opts.selectEnvironments(selectMoreEnvs)
+}
+
+func (opts *InitPipelineOpts) getEnvironments() ([]*archer.Environment, error) {
+	project := viper.GetString(projectFlag)
+	envs, err := opts.envStore.ListEnvironments(project)
```

Comment:
What do you think about storing these results in the struct?

The first time `getEnvironments` is called it fetches the full list and then writes it to `opts.projectEnvs`.
The second time `getEnvironmnets` is called it checks if `opts.projectEnvs != nil` and returns it.

Reply:
Yeah, I thought about whether it made sense to cache to save a call to SSM each time -- I wasn't sure if there would be any inconsistency if an env is added in the meantime and then it wouldn't get picked up, but now that I think about it that probably wouldn't happen, right?

Reply:
oh good point, yeah I think that'd be a rare scenario. Even if it happens, they can modify the pipeline file afterwards to add the missing environment so I don't think it's a big deal :)